src/components/tutorial-card/helpers/format-tutorial-card.ts

```diff
@@ -1,32 +1,40 @@
 import { getTutorialSlug } from 'views/collection-view/helpers'
 import getReadableTime from 'components/tutorial-meta/components/badges/helpers'
 import {
+	CollectionLite,
 	ProductUsed,
 	TutorialLite as ClientTutorialLite,
 } from 'lib/learn-client/types'
 import { TutorialCardPropsWithId } from '../types'
 
+/**
+ * Formats tutorial data for use with the Tutorial Card component.
+ */
 export function formatTutorialCard(
 	tutorial: ClientTutorialLite,
 	/**
-	 * Optional collection slug, used for URL.
+	 * Optional collection context.
+	 * This is used for the tutorial URL, as well as progress display.
+	 *
 	 * Note that each tutorial has a "default" collection context,
-	 * which we use if `collectionSlug` is not provided.
-	 * If `collectionSlug` is provided, it overrides that default
-	 * collection context. Note that `collectionSlug` is not validated;
+	 * which we use if `collectionContext` is not provided.
+	 *
+	 * If `collectionContext` is provided, it overrides that default
+	 * collection context. Note that `collectionContext` is not validated;
 	 * please ensure it corresponds to a valid collection.
 	 */
-	collectionSlug?: string
+	collectionContext?: Pick<CollectionLite, 'id' | 'slug'>
```

Comment:
We continue to require `CollectionLite` data on the `tutorial` argument for `format-tutorial-card`. This is necessary to display progress, as we need the collection `id` to query progress.

I've modified the second argument, which allows for overriding the default collection context, in order to also require `{ slug, id }` for the "alternate" collection context (rather than just `{ slug }` as this was previously).

Reply:
Question: Can a tutorial have different progress states across collections? 💭